### PR TITLE
Trailing whitespace discouraged

### PR DESCRIPTION
**Pull request by tychonievich**
One possible solution to #91

### CODE REVIEW

File: specification/gedcom.md [hunk 1]
```diff
@@ -316,6 +316,20 @@ Each non-pointer payload may be encoded in 1 or more line strings (line continua
 The exact encoding of non-pointer payloads is dependent on the datatype of the payload, as determined by the structure type.
 The datatype of non-pointer payloads cannot be fully determined by line value content alone.
 
+Because empty payloads and missing payloads are considered equivalent,
+both a structure with no payload
+and a structure with the empty string as its payload
+may be encoded either
+with an empty `LineVal` (i.e. with a space after the `Tag`)
+or with no `LineVal` (i.e. without a space after the `Tag`).
```
Comment (dthaler) [inline specification/gedcom.md:324]:
```suggestion
with an empty `LineVal` (i.e., with a space after the `Tag`)
or with no `LineVal` (i.e., without a space after the `Tag`).
```

File: specification/gedcom.md [hunk 2]
```diff
@@ -316,6 +316,20 @@ Each non-pointer payload may be encoded in 1 or more line strings (line continua
 The exact encoding of non-pointer payloads is dependent on the datatype of the payload, as determined by the structure type.
 The datatype of non-pointer payloads cannot be fully determined by line value content alone.
 
+Because empty payloads and missing payloads are considered equivalent,
+both a structure with no payload
+and a structure with the empty string as its payload
+may be encoded either
+with an empty `LineVal` (i.e., with a space after the `Tag`)
+or with no `LineVal` (i.e., without a space after the `Tag`).
+In both cases, it is recommended that the space be omitted.
```
Comment (tychonievich) [inline specification/gedcom.md:325]:
```suggestion
In both cases, it is recommended that the `LineVal` be omitted.
```

Comment (tychonievich) [reply specification/gedcom.md:325]: Inconsistent with spec; see #91 which says it's not discouraged, it's prohibited

File: specification/gedcom.md [hunk 3]
```diff
@@ -316,6 +316,24 @@ Each non-pointer payload may be encoded in 1 or more line strings (line continua
...
 The datatype of non-pointer payloads cannot be fully determined by line value content alone.
 
+Note that production `LineVal` does not match the empty string.
+Because empty payloads and missing payloads are considered equivalent,
+both a structure with no payload
+and a structure with the empty string as its payload
+are encoded with no `LineVal` and no space after the `Tag`.
+
+:::example
+The payload of a `MARR` structure has type `[Y|<NULL>]`, which is optional but cannot be the empty string.
```
Comment (dthaler) [inline specification/gedcom.md:326]:
```suggestion
The payload of a `MARR` structure has type `[Y|<NULL>]`, which is optional but if present cannot be the empty string.
```

File: specification/gedcom.md [hunk 4]
```diff
@@ -316,6 +316,24 @@ Each non-pointer payload may be encoded in 1 or more line strings (line continua
...
+The payload of a `EVEN` structure has type `Text`, which is not optional but can be the empty string.
+The `Line` encoding a no-payload `MARR` is "`1 MARR`"
+and the `Line` encoding an empty-payload `EVEN` is "`1 EVEN`";
+both `Line`s have no `LineVal` and no trailing space.
+:::
+
+:::note
+Implementations may choose to be more permissive and parse lines ending with a `Tag` and a single space as if they had no `LineVal`, but are not required to do so. They must not create lines of this form.
+:::
```
Comment (dthaler) [inline specification/gedcom.md:335]: I'm ok with this note, but I don't think it's necessary and would prefer removing it.

Comment (dthaler) [reply specification/gedcom.md:335]: Rationale for removal is that there's lots of other ways throughout the spec that implementations may choose to be more permissive and parse lines that should never occur, but we don't have notes for them.

Comment (tychonievich) [reply specification/gedcom.md:335]:
```suggestion
```

**Pull request merged**